### DISCUSSION TITLE
Is there a way to mutate input value before validation?

### DISCUSSION THREAD

**Discussion by ariaieboy** [filamentphp/filament#8227] [Help]
Hi, I want to know if is there any method to mutate input value before validation?

**Comment by dissto** [comment]
https://filamentphp.com/docs/3.x/actions/prebuilt-actions/create#lifecycle-hooks

Like this?

**Reply by ariaieboy** [reply]
It's for actions I need sth for TextInput

**Reply by leandrocfe** [reply]
> It's for actions I need sth for TextInput

beforeValidate?

https://filamentphp.com/docs/3.x/panels/resources/creating-records/#lifecycle-hooks

What would you like to do?

**Reply by ariaieboy** [reply]
This method runs on form lifecycle
I need to mutate inputs individually in my form schema
I have this problem with https://github.com/filamentphp/filament/issues/8238 masking an input.
and one solution that came to my mind was mutating the input before the validation step.

**Reply by djamesfar** [reply]
Looking for the same thing.  I usually use the prepareForValidation() method in Laravel form requests, eg. setting the slug based on name (don't show slug field in form) or setting required created_by user id from the auth()->id value.

**Reply by bt760** [reply]
You can override the  prepareForValidation() method in Pages Action and also apply that change to your field in mutateFormDataBeforeCreate() in Create action and mutateFormDataBeforeSave() in Edit action. For example this is my approach  in order to fix price mask problem in editing record

    protected function prepareForValidation($attributes)
    {
        $attributes['data']['price'] = $this->fixPriceMask($attributes['data']['price']);
        return $attributes;
    }

    protected function mutateFormDataBeforeSave(array $data): array
    {
        $data['price'] = $this->fixPriceMask($data['price']);
        return parent::mutateFormDataBeforeSave($data);
    }


    private function fixPriceMask($price): string
    {
        return str_replace(
            search: ',',
            replace: '',
            subject:$price
        );

    }

**Comment by codewithluck** [comment]
The prepareForValidation method  works for on form, but it does not work with TextInput of the table list. Is there anyway to modify the TextInput state of the table before running validation?
